src/components/engines/cms/modules/media/MediaGrid.tsx

```
@@ -64,9 +128,30 @@ export function MediaGrid({
   };
+
+  // Filter items to show only visible ones
+  const itemsToShow = items.filter(item => visibleItems.includes(item.id));
   
   return (
     <div className="p-4 min-h-[400px]">
+      {/* Loading indicator */}
+      {visibleItems.length < items.length && (
+        <div className="mb-4 text-center">
+          <div className="inline-flex items-center px-4 py-2 text-sm text-blue-600 bg-blue-50 rounded-full border border-blue-200">
+            <div className="w-4 h-4 mr-3 border-2 border-blue-600 border-t-transparent rounded-full animate-spin"></div>
```

Biome (1.9.4)

[error] JSX elements without children should be marked as self-closing. In JSX, it is valid for any element to be self-closing.

Unsafe fix: Use a SelfClosingElement instead


(lint/style/useSelfClosingElements)